src/components/AppMapDetailsObj.ts

```diff
@@ -415,4 +418,100 @@ export default class AppMapDetailsObj extends AppMapDetailsBase<MapMarkerObj | M
     }
     return "Normal";
   }
+  korokType() {
+    //if (!this.obj || !('korok_id' in this.obj)) {
+    //  return "NaK";
+    //}
+    return this.obj && this.obj.korok_type == 'Flower Trail';
+  }
+
+  findItemByHash(group: any[], links: any[], name: string): any {
+    let hashes = links.map(link => link.DestUnitHashId);
+    let out = group.filter(g => g.data.UnitConfigName == name && hashes.includes(g.hash_id));
```

Comment:
You can use [Array.find()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead (but return null, not undefined, if find() returns undefined).